src/sagemaker/deserializers.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2017-2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""Implements methods for deserializing data returned from an inference endpoint."""
+from __future__ import absolute_import
+
+import abc
+
+
+class BaseDeserializer(abc.ABC):
+    """Abstract base class for creation of new deserializers.
+
+    Provides a skeleton for customization requiring the overriding of the method
+    deserialize and the class attribute ACCEPT.
+    """
+
+    @abc.abstractmethod
+    def deserialize(self, data, content_type):
+        """Deserialize data received from an inference endpoint.
+
+        Args:
+            data (object): Data to be deserialized.
+            content_type (str): The MIME type of the data.
+
+        Returns:
+            object: The data deserialized into an object.
+        """
+        raise NotImplementedError
+
+    @property
+    @abc.abstractmethod
+    def ACCEPT(self):
```

Comment:
nit-pep8: [lowercase](https://www.python.org/dev/peps/pep-0008/#function-and-variable-names) -- only [constants](https://www.python.org/dev/peps/pep-0008/#constants) should be all caps:

```suggestion
    def accept(self):
```

Reply:
ACCEPT is a constant. The use of @property with @abstractmethod is confusing, because it suggests that the user needs to implement a method rather than a class-level constant. But, defining a class-level constant also satisfies the interface. 

For example, the following code satisfies the `BaseSerializer` interface:
```
class FooDeserializer(BaseDeserialzier):
    ACCEPT = "application/json"
   
    def deserialize(self, data):
        pass
```

This is how I'm planning on implementing all the native serializers/deserializers, and also how I intend to encourage users to implement custom serializer. My reasons are two-fold:

1) I assume that there would be no reason to change accept and content-types at runtime, so it makes sense to specify them as a constant.
2) Using a class-level constant is a pattern seen in other Python packages (e.g., [see this PyTorch object](https://pytorch.org/tutorials/intermediate/pruning_tutorial.html#extending-torch-nn-utils-prune-with-custom-pruning-functions)).

Reply:
makes sense re: class-level constants. my nit comments were more about the violating naming convention for the method in the base class, as it wasn't clear that your intent/desire was to have the derived classes implement the property by defining a constant.